backend/src/agents/adapters.py

```diff
@@ -1,22 +1,54 @@
+from typing import List
+
 from .tool import Tool
+import json
+
+def create_all_tools_str(tools: List[Tool]) -> str:
+    return "".join(to_object(tool) + "\n\n" for tool in tools)
 
 
-def convert_to_mistral_tool(tool: Tool):
-    return {
-        "type": "function",
-        "function": {
-            "description": tool.description,
-            "name": tool.action.__name__,
-            "parameters": {
-                "type": "object",
-                "properties": {
-                    key: {
-                        "type": inner_dict.type,
-                        "description": inner_dict.description,
-                    }
-                    for key, inner_dict in tool.parameters.items()
-                },
-                "required": [key for key, inner_dict in tool.parameters.items() if inner_dict.required],
-            },
+def to_object(tool: Tool) -> str:
+    obj = {
+        "description": tool.description,
+        "name": tool.name,
+        "parameters": {
+            key: {
+                    "type": inner_dict.type,
+                    "description": inner_dict.description,
+                }
+                for key, inner_dict in tool.parameters.items()

```

Comment:
Should this method live on every tool class?

Reply:
I'm confused what you mean. The `to_object()` method is exclusively used in `create_all_tools_str()` and that is a dependency on the `agent.py` invoke method itself.

Are you suggesting we move it to the `tool.py` file? I don't believe we have a tool class per se but if we did it would be a good place to house this

Reply:
Yeah the latter, I mean should / can it live on this class ? https://github.com/WaitThatShouldntWork/InferGPT/blob/8e135eb5964700ecf9558b60b6dc4e20e5bbec17/backend/src/agents/types.py#L20

Reply:
I have made these changes as discussed on call in [e5082d7](https://github.com/WaitThatShouldntWork/InferGPT/pull/55/commits/e5082d744aaedd9407e8a7e0d9c0defd61853e1e)